src/decaylanguage/dec/dec.py

```diff
@@ -196,11 +196,22 @@ def parse(self, include_ccdecays: bool = True) -> None:
         parser = Lark(
             self.grammar(), parser=opts["parser"], lexer=opts["lexer"], **extraopts
         )
-
         self._parsed_dec_file = parser.parse(self._dec_file)
 
+        # Strip whitespace and semicolons (necessary for LALR(1) grammar) from model names
+        self._parsed_dec_file = ModelNameCleanup().transform(self._parsed_dec_file)
```

Comment:
This is the first of 2 transformers. For small files there's no issue but for master .DEC files it may add some non-negligible time to navigate twice. Let's see as I've just asked you to time a couple of examples ... If timing increases non-negligibly, would it be possible to do these 2 transformers in one? Fine if this goes as a future endeavour - I'm also very keen on seeing this PR go in and released.

Reply:
I've tested this now and there is some significant slowdown:

LHCb: 18% slower
Belle II: 43% slower (using the old decay file)

For us being able to use this tool at all is great so I wouldn't see this as unacceptable but I see that the LHCb perspective might be somewhat different here.

Combining the two transformers is not straightforward as the first transformer also replaces the semicolon / whitespace in the second one. I'm sure there are ways to decrease the performance penalty but that might require some time on my end.

Reply:
Thanks a lot for the speedy reaction.

OK, it does have a clear impact but not the end of the world either. We can probably work on improving in the future ...

I guess the larger difference for the Belle II file is simply stating that model aliases add a clear time penalty in parsing.

No worries, this is not unacceptable at all. I should say that I started on this really for anyone using dec file, not specifically for LHCb. In that sense, to me it is *extremely important and relevant* to have Belle II colleagues on board. If you are happy then I'm happy that this tool can serve us all :-).

Totally understand that the merging is not straightforward. I realised that after making my comment. No worries.

Good that we have these numbers in mind for any future reference.